tidy3d/components/tcad/boundary/abstract.py

```diff
@@ -0,0 +1,11 @@
+"""Defines heat material specifications"""
+
+from __future__ import annotations
+
+from abc import ABC
+
+from tidy3d.components.base import Tidy3dBaseModel
+
+
+class HeatChargeBC(ABC, Tidy3dBaseModel):
+    """Abstract heat-charge boundary conditions."""
```

Comment:
If we're going to change the directory we might as well change the naming here to `TcadBC` or something along those lines?

Reply:
@momchil-flex what do you think about this directory name? Is it ok?

We can still make things like the `HeatSimulation` backwards compatible by just changing the import location at the tidy3d top level and making sure no internals change at all in the updated PR.

Reply:
I am not sure. On one hand I feel like TCAD is a bit too general here. Also when you think what it means, not referring to any specific physics but just to computer-aided design, it seems strange to call things e.g. `TcadBC`. On the other hand, it is widely used and understood to be centered around exactly what we're solving here. *Maybe* my suggestion would be to rename the folder while still keeping `HeatCharge` in component names? Does that make sense?

Reply:
Yep nice, I agree